src/main/java/com/yilmaznaslan/AppointmentFinder.java

```diff
@@ -98,35 +100,34 @@ private void run() {
 
 
         } catch (UnhandledAlertException e) {
-            LOGGER.error("UnhandledAlertException occurred during getting the home page.Clicking", e);
+            logger.error("UnhandledAlertException occurred during getting the home page.Clicking", e);
             Alert alert = driver.switchTo().alert();
             alert.accept();
         } catch (NoSuchSessionException e) {
-            LOGGER.error("NoSuchSessionException occurred during getting the home page. Creating a new Driver instance", e);
+            logger.error("NoSuchSessionException occurred during getting the home page. Creating a new Driver instance", e);
             driver = DriverUtils.initDriver();
         } catch (InterruptedException e) {
             Thread.currentThread().interrupt();
-            LOGGER.error("Interrupted: ", e);
+            logger.error("Interrupted: ", e);
         } catch (Exception e) {
-            LOGGER.error("General Exception: ", e);
+            logger.error("General Exception: ", e);
             IoUtils.savePage(driver, "exception_section2");
         }
 
     }
 
-    private void getHomePage() {
+    private void loadHomePage(RemoteWebDriver driver, String url, Logger logger) {
```

Comment:
+ Nice renaming.
 
these parameters are really not needed. They are already available as class variables anyway.  Especially Logger .I must say I have never seen something like this :)

Reply:
I added those parameters to make the code testable.

As is now, it is impossible to test that method in isolation.
My mocking those parameters, it can be done with ease.

I suggest you check some introductory resources on refactoring and testing.

I can think of: https://refactoring.guru/refactoring

In case you are interested in more pointers about this, just let me know in private.

Reply:
I don't think it is impossible to test it in isolation it is isolated anyway.  It just needs to be protected to be able to access it in unit tests

Reply:
So your objection is to the access modifier, and not to the new parameters I added?

Reply:
> I don't think it is impossible to test it in isolation it is isolated anyway. It just needs to be protected to be able to access it in unit tests

Could you please provide me with a working unit test that uses the previous form of the method?

Reply:
> It just needs to be protected to be able to access it in unit tests

I think you're right about the permissions, though.